src/prefect/agent/agent.py

```diff
@@ -207,9 +207,23 @@ def deploy_and_update_flow_run(self, flow_run: "GraphQLResult") -> None:
         """
         # Deploy flow run and mark failed if any deployment error
         try:
+            self.update_state(flow_run)
             deployment_info = self.deploy_flow(flow_run)
-            self.update_state(flow_run, deployment_info)
+            self.client.write_run_logs(
+                [
+                    dict(
+                        flowRunId=getattr(flow_run, "id", "UNKNOWN"),  # type: ignore
```

Comment:
I believe this will have an error when `UNKNOWN` and then the catch will still try to write a log again w/ `UNKNOWN` and fail.

Reply:
@joshmeek updated